provider/common.go

```diff
@@ -96,12 +97,10 @@ func (rt *resourceTemplate) escape(name string) (string, error) {
 	if err != nil {
 		return "", err
 	}
-	name = reg.ReplaceAllString(name, "-")
-	if strings.HasSuffix(name, "-") {
-		name = name[:len(name)-1]
-	}
-	if strings.HasPrefix(name, "-") {
-		name = name[1:]
+	name = reg.ReplaceAllString(name, "_")
+	name = strings.Trim(name, "_")
+	if unicode.IsDigit(rune(name[0])) {
+		name = fmt.Sprintf("_%s", name)
```

Comment:
a terraform resource name cannot start with a digit... I think it is an arbitrary rule decided by Hashicorp folks.